python/ray/dag/constants.py

```diff
@@ -4,14 +4,15 @@
 PARENT_CLASS_NODE_KEY = "parent_class_node"
 PREV_CLASS_METHOD_CALL_KEY = "prev_class_method_call"
 BIND_INDEX_KEY = "bind_index"
-NO_CONTROL_EDGE_BIND_INDEX_VALUE = -1
+WRITE_BIND_INDEX_INCREMENT = 0.3
```

Comment:
This is hacky. Can you assign consecutive bind indicies to nccl read/write and the corresponding compute nodes?

Reply:
Ideally, we do want consecutive bind indices:
- If the task represented by a `ClassMethodNode` requires NCCL read from one or many upstream tasks, for each of its corresponding `_NcclRecvNode`(s), `_NcclRecvNodes[i].bind_index == ClassMethodNode.bind_index - i - 1`.
- If the task represented by a `ClassMethodNode` requires NCCL write, its corresponding `_NcclSendNode.bind_index == ClassMethodNode.bind_index + 1`.

Under the current implementation, it seems not possible to assign consecutive bind indices at bind time. This is because the user can use `with_type_hint` to add/remove type hints.

Reply:
Conceptually, the NCCL read/write tasks are still subtasks of the original task, so it makes sense to subtract a little for NCCL read and add a little for NCCL write 🤣 . The problem is that the value added/subtracted can be arbitrary.

Another idea would be to go through all actor tasks in the DAG and update all of their bind indices. I feel this requires a lot of code change and might be confusing.

Another idea I have tried is to set the bind indices of all NCCL reads and writes to an invalid value -1. This messes up the `exec_task_idx`s and generates correct but non-readable schedules (because of the `exec_task_idx`s).

Another idea is to set the bind index of all NCCL reads and write to be the same as their corresponding `ClassMethodNode`'s bind index. This can still result in arbitrary `exec_task_idx`s because of different ways of tie-breaking.

Reply:
Use the same bind index. Change sort lambda.